components/MuseumTile.tsx

```diff
@@ -1,39 +1,112 @@
-import {View, Text, ActivityIndicator, StyleSheet} from "react-native";
+import {
+    View,
+    Text,
+    ActivityIndicator,
+    StyleSheet,
+    TouchableOpacity,
+    useColorScheme,
+    TouchableHighlight
+} from "react-native";
 import getMuseum from "../api/getMuseum";
 import FontAwesome from '@expo/vector-icons/FontAwesome';
+import {useNavigation} from "@react-navigation/native";
+import {MainStackNavigation} from "../App";
+import {Image} from 'expo-image';
+import {Appearance} from 'react-native';
+import {MOCK_IMAGE} from "../constans";
+import {
+    darkBackground,
+    darkHighlight,
+    darkSurface,
+    darkText,
+    lightBackground,
+    lightHighlight,
+    lightSurface,
+    lightText
+} from "../colors";
+import useAppColorScheme from "../hooks/useAppColorScheme";
 
 interface MuseumTileProps {
     id: number;
     selected: boolean,
-    onTap: Function,
+    onFavouriteTap: Function,
 }
 
 export default function MuseumTile(props: MuseumTileProps) {
+    const colorScheme = useAppColorScheme();
 
     const museumAction = getMuseum(props.id);
+    const navigation = useNavigation<MainStackNavigation>();
+
+    const onTileTap = () => {
+        if (!museumAction.inProgress) {
+            navigation.navigate("DetailedMuseum", {"museum": museumAction.data!})
```

Comment:
Just a minor thing, since it's not a JSON :) 

```suggestion
            navigation.navigate("DetailedMuseum", {museum: museumAction.data!})
```